File: 0232-implement-queue-using-stacks/0232-implement-queue-using-stacks.java
class MyQueue {

    Stack<Integer> s1 = new Stack<Integer>();
    Stack<Integer> s2 = new Stack<Integer>();

    public MyQueue() {
        
    }
    public void push(int x) {
        s1.push(x);
        
    }
    public int pop() {
        if(s2.empty()){
            while(!s1.empty()){
                int x = s1.pop();
                s2.push(x);
            }
        }
       int res = s2.pop();
        if(s1.empty()){
            while(!s2.empty()){
                int x = s2.pop();
                s1.push(x);
            }
        }
       return res;    
    }
    public int peek() {
        if(s2.empty()){
            while(!s1.empty()){
                int x = s1.pop();
                s2.push(x);
            }
        }
       int res = s2.peek();
        if(s1.empty()){
            while(!s2.empty()){
                int x = s2.pop();
                s1.push(x);
            }
        }
       return res;   
    }
    public boolean empty() {
        return s1.empty();
    }
}

File: 0232-implement-queue-using-stacks/0232-implement-queue-using-stacks.cpp
class MyQueue {
public:
    stack<int> s;
    stack<int> s2;
    
    MyQueue() {
        
    }
    
    void push(int x) {
        s.push(x);
    }
    
    int pop() {
        while(!s.empty()){
            s2.push(s.top());
            s.pop();
        }
        int elem = s2.top();
        s2.pop();
        while(!s2.empty()){
            s.push(s2.top());
            s2.pop();
        }
        return elem;
    }
    
    int peek() {
        while(!s.empty()){
            s2.push(s.top());
            s.pop();
        }
        int elem = s2.top();
        while(!s2.empty()){
            s.push(s2.top());
            s2.pop();
        }
        return elem;
    }
    
    bool empty() {
        return s.empty();
    }
};

/**
 * Your MyQueue object will be instantiated and called as such:
 * MyQueue* obj = new MyQueue();
 * obj->push(x);
 * int param_2 = obj->pop();
 * int param_3 = obj->peek();
 * bool param_4 = obj->empty();
 */